### PR TITLE
Fetcher retries

### PR DESCRIPTION
**Pull request by barbarahui**
- implement retries in base Fetcher class
- move `make_http_request()` functionality into `ucd_json_fetcher` -- this is the only place this was being used

### CODE REVIEW

File: metadata_fetcher/fetchers/Fetcher.py [hunk 1]
```diff
@@ -44,11 +44,22 @@ def __init__(self, params: dict):
...
+        self.http_session = self.configure_http_session()
 
         if not self.collection_id:
             raise CollectionIdRequired("collection_id is required")
 
+    def configure_http_session(self) -> requests.Session:
+        http = requests.Session()
+        retry_strategy = Retry(
+            total=3,
+            status_forcelist=[413, 429, 500, 502, 503, 504],
```
Comment (amywieliczka) [inline metadata_fetcher/fetchers/Fetcher.py:56]: Why did you remove the back off factor - that seems useful and relevant?

File: metadata_fetcher/fetchers/ucd_json_fetcher.py [hunk 2]
```diff
@@ -89,7 +90,10 @@ def fetch_json_ld(self, url: str) -> Optional[dict]:
         Returns:
             dict
         """
-        response = self.make_http_request(url)
+        session = requests.Session()
+        retries = Retry(total=3, backoff_factor=2)
+        session.mount("https://", HTTPAdapter(max_retries=retries))
+        response = session.get(url=url)
```
Comment (amywieliczka) [inline metadata_fetcher/fetchers/ucd_json_fetcher.py:96]: If you added the backoff_factor to the gloabl configuration for `self.http_session`, then you should be able to do: 

```suggestion
        response = self.http_session.get(url=url)
```

And get rid of lines 93-95 here. 

Sure, `self.http_session` adds some status codes to the forcelist and attached the retry strategy to both http and https, but that seems fine?

File: metadata_fetcher/fetchers/Fetcher.py (hunk 1 above)
Comment (amywieliczka) [reply metadata_fetcher/fetchers/Fetcher.py:56]: I also wonder if this makes sense as a global configuration - like in `rikolti/utils.py`?

**Review by amywieliczka — CHANGES_REQUESTED** [review]

**Comment by amywieliczka** [comment]
Looks like `nuxeo_fetcher`, `oac_fetcher`, and `ucd_json_fetcher` all make their own calls to `requests.get`, so you might want to update those other instances in `nuxeo_fetcher` and `oac_fetcher` to use `self.http_session.get` instead.

File: metadata_fetcher/fetchers/Fetcher.py (hunk 1 above)
Comment (barbarahui) [reply metadata_fetcher/fetchers/Fetcher.py:56]: Well, I didn't actually remove the backoff factor as there wasn't one configured at all here previously. I figured that we could go with the default backoff factor of 0 and tweak if necessary. But sure, it probably makes sense to set it to `2`.

Yeah, we could make this a global configuration. This is actually a copy of `configure_http_session()` from the content harvester code (which doesn't have a backoff factor set).

File: metadata_fetcher/fetchers/ucd_json_fetcher.py (hunk 2 above)
Comment (barbarahui) [reply metadata_fetcher/fetchers/ucd_json_fetcher.py:96]: Yeah, I thought about that too. I figured to introduce the least amount of change possible, but you're right, it probably would be fine.

**Comment by barbarahui** [comment]
@amywieliczka Sure yeah, I think it makes sense to do this. I can do the work to put this in place. I don't think it'll take too long--I was just trying to do this quickly so that I could focus on the Nuxeo API issue.

File: metadata_fetcher/fetchers/ucd_json_fetcher.py (hunk 2 above)
Comment (amywieliczka) [reply metadata_fetcher/fetchers/ucd_json_fetcher.py]: Mmmm, that makes sense - what you did change (`Fetcher.fetch_page()`) is the function that is called like 95% of the time though, so it seemed confusing to me to change it most of the way but not all of the way. Could result in weird errors on strange edge cases where we've had to implement additional requests outside of the standard requests managed by the Fetcher base class.

**Comment by amywieliczka** [comment]
Figured I could do this quickly while @barbarahui's head was in Nuxeo API issues. 
Resolved my own asks - @barbarahui I'll leave this open in case you want to take a look, but as far as I'm concerned it's good to go - rebase & merge when you get to it.

**Comment by barbarahui** [comment]
@amywieliczka this looks great, thank you so much!!!!
